nodeup/pkg/model/ec2instanceconnect.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"k8s.io/klog"
+	"k8s.io/kops/nodeup/pkg/distros"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+)
+
+// DockerBuilder install docker (just the packages at the moment)
+type EC2InstanceConnectBuilder struct {
+	*NodeupModelContext
+}
+
+var _ fi.ModelBuilder = &EC2InstanceConnectBuilder{}
+
+// Build is responsible for installing packages
+func (b *EC2InstanceConnectBuilder) Build(c *fi.ModelBuilderContext) error {
+	if !b.InstanceGroup.Spec.EC2InstanceConnect {
+		return nil
+	}
+
+	if b.Distribution.IsUbuntu() || b.Distribution == distros.DistributionAmazonLinux2 {
+		c.AddTask(&nodetasks.Package{Name: "ec2-instance-connect"})
+	} else if b.Distribution.IsDebianFamily() {
```

Comment:
This one is a bit flaky. I need to remove support for Debian unless we host this deb somewhere else. Ubuntu is quite fond of breaking deb paths, it seems.